flutter/lib/src/navigation/sentry_navigator_observer.dart

```diff
@@ -100,6 +116,32 @@ class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
       });
     }
   }
+
+  void _startTransaction(String? name, Object? arguments) {
+    if (!_enableAutoTransactions) {
+      return;
+    }
+    if (name == null) {
+      return;
+    }
+    _hub.configureScope((scope) {
+      _transaction = _hub.startTransaction(
+        name,
+        'ui.load',
+        bindToScope: scope.span == null,
+        waitForChildren: true,
+      );
+      if (arguments != null) {
+        _transaction?.setData('route_settings_arguments', arguments);
+      }
+      _transaction?.finishAfter(Duration(seconds: 3), status: SpanStatus.ok());
+    });
+  }
+
+  Future<void> _finishTransaction() async {
+    _transaction?.status ??= SpanStatus.ok();
+    return await _transaction?.finish();
```

Comment:
Is there a reason you're setting the status manually instead as a parameter of `finish()`?

Reply:
Yeah because we only set the status if none was previously set with `??=` operator.